pkg/motor/resource.go

```diff
@@ -0,0 +1,111 @@
+package motor
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"sync"
+
+	"github.com/sonr-io/sonr/pkg/client"
+	st "github.com/sonr-io/sonr/x/schema/types"
+)
+
+// SchemaCallback is a callback used when a schema CID is fetched
+type SchemaCallback func(*st.SchemaDefinition, error)
+type motorResources struct {
+	mu sync.Mutex
+
+	config            *client.Client
+	schemaQueryClient st.QueryClient
+
+	whatIsStore map[string]*st.WhatIs
+	schemaStore map[string]*st.SchemaDefinition
+}
+
+func newMotorResources(config *client.Client, schemaQueryClient st.QueryClient) *motorResources {
+	return &motorResources{
+		config:            config,
+		schemaQueryClient: schemaQueryClient,
+
+		whatIsStore: make(map[string]*st.WhatIs),
+		schemaStore: make(map[string]*st.SchemaDefinition),
+	}
+}
+
+/// StoreWhatIs fetches
+func (r *motorResources) StoreWhatIs(whatIs *st.WhatIs, cb SchemaCallback) {
+	callback := func(s *st.SchemaDefinition, e error) {
+		r.mu.Unlock()
+		cb(s, e)
+	}
+
+	go func() {
```

Comment:
im a bit confused on why this is running in a go routine. is this to load the schema in a background context? it might be better to load in the same context and invoke the callback in the same execution context.

Reply:
i think we should introduce it later if we need this kind of implementation. im in favor of the callback approach but think execution should stay synchronous

Reply:
sgtm, done